src/hash_witness.rs

```diff
@@ -85,34 +85,20 @@ impl HashName for ConsName {
         match self {
             Self::NeverUsed => MAX_CONSES_PER_REDUCTION + 1,
             Self::Expr => 0,
-            Self::ExprCdr => 1,
-            Self::UnevaledArgsCdr => 1,
+            Self::ExprCdr | Self::UnevaledArgsCdr => 1,
             Self::ExprCadr => 2,
-            Self::ExprCddr => 3,
-            Self::UnopConsLike => 3,
-            Self::Lambda => 3,
-            Self::ExprCaadr => 4,
-            Self::Begin => 4,
-            Self::InnerBody => 4,
-            Self::ExtendedClosureEnv => 4,
-            Self::UnevaledArgs => 5,
-            Self::ExprCaaadr => 5,
-            Self::ExtendedRec => 5,
-            Self::EnvToUse => 5,
-            Self::Binding => 5,
-            Self::FunBody => 6,
-            Self::NewRecCadr => 6,
-            Self::NewRec => 7,
-            Self::ClosedEnv => 7,
-            Self::Env => 8,
-            Self::ExpandedInner0 => 8,
-            Self::FunExpanded => 9,
-            Self::Expanded => 9,
-            Self::EnvCar => 9,
-            Self::InnerLambda => 10,
-            Self::TheCons => 10,
-            Self::EnvCaar => 10,
-            Self::ExpandedInner => 10,
+            Self::ExprCddr | Self::UnopConsLike | Self::Lambda => 3,
+            Self::ExprCaadr | Self::Begin | Self::InnerBody | Self::ExtendedClosureEnv => 4,
+            Self::UnevaledArgs
+            | Self::ExprCaaadr
+            | Self::ExtendedRec
+            | Self::EnvToUse
+            | Self::Binding => 5,
+            Self::FunBody | Self::NewRecCadr => 6,
+            Self::NewRec | Self::ClosedEnv => 7,
+            Self::Env | Self::ExpandedInner0 => 8,
+            Self::FunExpanded | Self::Expanded | Self::EnvCar => 9,
+            Self::InnerLambda | Self::TheCons | Self::EnvCaar | Self::ExpandedInner => 10,
```

Comment:
This is one instance in which I would not enforce this stylistic change. Expressing the information this way would have made it much harder (or in practical terms, impossible) to reach this state; and it will make any hypothetical future changes (for bug fixes or when extending/changing the structure of reduction) much harder. The fundamental issue is that the duplicated matches are coincidental: they're really only associated because they *don't* have any relationship to each other and so can share a slot. Requiring the code to express the incidental relationship concretized in a particular layout will make adjusting and optimizing those layouts more difficult.

I'm fine with the lint generally, but let's revert this example and use an exception.